helm-charts/konk-service/templates/apiservice-configmap.yaml

```diff
@@ -30,6 +30,25 @@ data:
     fi
     echo 'Done' > /tmp/healthy
 
+  delete-api-service.sh: |
+    #!/bin/bash
+    set -xe
+
+    CERT=$(cat certs/ca.crt | base64)
+    CERT=${CERT//$'\n'/}
+    cat /mounts/api-service.yaml.in | sed "s/{{` {{ SERVICENAME }} `}}/${SERVICENAME}/g" | sed "s/{{` {{ NAMESPACE }} `}}/${NAMESPACE}/g" | sed "s/{{` {{ CERT }} `}}/${CERT}/g" > /gen/api-service.yaml
+    delay=2
+    until kubectl delete -f /gen/api-service.yaml
+    do
+      sleep $delay
+      let delay="delay*2"
+      if [ "$delay" -gt "60" ]
+      then
+        delay=60
+      fi
+    done
```

Comment:
You might not want to retry here. If it fails, it could be that the konk has been deleted, in which case there's no longer a need to delete the API service. Or maybe the APIService has already been deleted by something else.

Reply:
Removing the retry and converting the pod to a job.